File: src/main/java/com/leijendary/spring/boot/template/api/v1/data/SampleRequest.java
package com.leijendary.spring.boot.template.api.v1.data;

import com.leijendary.spring.boot.template.validator.annotation.v1.FieldsNotEqualV1;
import lombok.Data;

import javax.validation.Valid;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.Size;
import java.util.HashSet;
import java.util.Set;

@Data
@FieldsNotEqualV1
public class SampleRequest {

    @NotBlank(message = "validation.required")
    @Size(max = 50, message = "validation.maxLength")
    private String field1;

    private int field2;

    @Valid
    @NotEmpty(message = "validation.required")
    public Set<SampleTranslationRequest> translations = new HashSet<>();
}

File: src/main/kotlin/com/leijendary/spring/boot/template/api/v1/data/SampleRequest.kt
package com.leijendary.spring.boot.template.api.v1.data

import java.math.BigDecimal
import java.math.BigDecimal.ZERO
import javax.validation.Valid
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.NotBlank
import javax.validation.constraints.NotEmpty
import javax.validation.constraints.Size

data class SampleRequest(
    @get:NotBlank(message = "validation.required")
    @get:Size(max = 50, message = "validation.maxLength")
    val field1: String? = null,

    val field2: Long = 0,

    @get:DecimalMin(value = "0.00", message = "validation.decimal.min")
    val amount: BigDecimal = ZERO,

    @get:Valid
    @get:NotEmpty(message = "validation.required")
    val translations: MutableSet<SampleTranslationRequest>? = HashSet()
)